pkg/container-hook/tracer.go

```diff
@@ -432,12 +463,12 @@ func (n *ContainerNotifier) watchPidFileIterate(
 	// See fix in Linux 5.9:
 	// https://github.com/torvalds/linux/commit/497b0c5a7c0688c1b100a9c2e267337f677c198e
 	// Workaround: remove parent mask. We don't need it anymore :)
-	err = pidFileDirNotify.Mark(unix.FAN_MARK_REMOVE, unix.FAN_ACCESS_PERM|unix.FAN_EVENT_ON_CHILD, unix.AT_FDCWD, pidFileDir)
+	err = n.pidFileDirNotify.Mark(unix.FAN_MARK_REMOVE, unix.FAN_ACCESS_PERM|unix.FAN_EVENT_ON_CHILD, unix.AT_FDCWD, pc.pidFileDir)
```

Comment:
Does it also remove the ignore marks added in L561 and L574?

Reply:
Also, should it be done before in the code to avoid "leaking" this mark in case of error?

Reply:
> Does it also remove the ignore marks added in L561 and L574?

Good finding. No, it does not remove the ignore marks, unless the files with the ignore marks are deleted. This bug existed before this PR.

> Also, should it be done before in the code to avoid "leaking" this mark in case of error?

It depends on the error. `watchPidFileIterate()` is called for all FAN_ACCESS_PERM events in files located in the pid file directory. If the file is not related to the pid file we want to monitor, we don't want to release the marks yet.

Also, if `runc create` fails, the pid file will not be created. Thanks to this PR, we're not leaking any fanotify fd, but we're still leaking the marks at the moment...